python/imageutils.py

```diff
@@ -212,6 +212,70 @@ def cp_block_block(self, legend):
                     #    end_z = 0
                     #prev_b_end = (block_y + shape[0], block_z + end_z, block_x + end_x)
                     #print "End position of write: {0}".format(prev_b_end)
+    
+    #will only work if the amount of blocks that fit in memory represents a rectangular prism
+    def cp_block_slice(self, legend, mem):
+        with open(self.filepath, "r+b") as reconstructed:
+
+            rec_dims = self.proxy.header.get_data_shape()
+
+            y_size = rec_dims[0]
+            z_size = rec_dims[1]
+            x_size = rec_dims[2]
+
+            with open(legend, "r") as legend:
+
+                remaining_mem = mem
+                data = None
+                for block_name in legend:
+                    
+                    block_name = block_name.strip()
+                    print "Reading block {0}".format(block_name)
+
+                    block_pos = block_name[:-4].split('_')
+                    start_x = 0
```

Comment:
This is a bug and should not be here, but rather outside of the for loop. Currently fixing it.